geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/AccumuloVersion.scala

```diff
@@ -56,7 +58,37 @@ object AccumuloVersion extends Enumeration {
     }
   }
 
-  def getTypeFromClass[T: ClassTag](className: String, field: String): T = {
+  def ensureTableExists(connector: Connector, table: String): Unit = {
+    val tableOps = connector.tableOperations()
+    if (!tableOps.exists(table)) {
+      try {
+        ensureNamespaceExists(connector, table)
+        tableOps.create(table, true, TimeType.LOGICAL)
+      } catch {
+        // this can happen with multiple threads but shouldn't cause any issues
+        case e: TableExistsException =>
+        case e: Exception if e.getClass.getSimpleName == "NamespaceExistsException" =>
+      }
+    }
+  }
+
+  /**
+   * Check for namespaces and create if needed.
+   */
+  private def ensureNamespaceExists(connector: Connector, table: String): Unit = {
+    val dot = table.indexOf('.')
+    if (dot > 0) {
+      require(accumuloVersion != V15, s"Table namespaces are not supported in Accumulo 1.5 - for table '$table'")
```

Comment:
cool, yeah i guess i never tried that...but i do remember encoding non alpha+underscores in the table name at one point